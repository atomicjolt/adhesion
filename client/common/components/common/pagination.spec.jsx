import React from 'react';
import { shallow } from 'enzyme';

import Pagination from './pagination';

describe('Pagination', () => {
  const props = {
    changePageTo: () => {},
    currentPage: 2,
    previousPageAvailable: true,
    nextPageAvailable: true,
  };

  it('renders the pagination UI', () => {
    const result = shallow(<Pagination
      changePageTo={props.changePageTo}
      currentPage={props.currentPage}
      previousPageAvailable={props.previousPageAvailable}
      nextPageAvailable={props.nextPageAvailable}
    />);

    expect(result).toMatchSnapshot();
  });

  describe('when the user clicks the previous button', () => {
    it('navigates to the previous page', () => {
      spyOn(props, 'changePageTo');
      const result = shallow(<Pagination
        changePageTo={props.changePageTo}
        currentPage={props.currentPage}
        previousPageAvailable={props.previousPageAvailable}
        nextPageAvailable={props.nextPageAvailable}
      />);

      const previousButton = result.find('button').first();
      previousButton.simulate('click');

      expect(props.changePageTo).toHaveBeenCalledWith(
        props.currentPage - 1,
      );
    });
  });

  describe('when the user clicks the next button', () => {
    it('navigates to the next page', () => {
      spyOn(props, 'changePageTo');
      const result = shallow(<Pagination
        changePageTo={props.changePageTo}
        currentPage={props.currentPage}
        previousPageAvailable={props.previousPageAvailable}
        nextPageAvailable={props.nextPageAvailable}
      />);

      const nextButton = result.find('button').last();
      nextButton.simulate('click');

      expect(props.changePageTo).toHaveBeenCalledWith(
        props.currentPage + 1,
      );
    });
  });

  describe('when there is no previous page available', () => {
    it('disables the previous button', () => {
      const result = shallow(<Pagination
        changePageTo={props.changePageTo}
        currentPage={props.currentPage}
        previousPageAvailable={false}
        nextPageAvailable={props.nextPageAvailable}
      />);

      const previousButton = result.find('button').first();

      expect(previousButton.prop('disabled')).toBe(true);
    });
  });

  describe('when there is no next page available', () => {
    it('disables the next button', () => {
      const result = shallow(<Pagination
        changePageTo={props.changePageTo}
        currentPage={props.currentPage}
        previousPageAvailable={props.previousPageAvailable}
        nextPageAvailable={false}
      />);

      const nextButton = result.find('button').last();

      expect(nextButton.prop('disabled')).toBe(true);
    });
  });
});
