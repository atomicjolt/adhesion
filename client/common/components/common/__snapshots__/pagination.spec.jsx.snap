// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders the pagination UI 1`] = `
<nav>
  <ul>
    <button
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      &lt;
    </button>
    <button
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      &gt;
    </button>
  </ul>
</nav>
`;
